app/src/androidTest/java/com/droidhats/campuscompass/views/MapFragmentTest.kt

```diff
@@ -54,12 +64,211 @@ class MapFragmentTest {
+                withId(R.id.search_info), withText("Enter Street, Address, Concordia Classroom..."),
+                isDisplayed()
+            )
+        )
+        infoTextView.check(matches(withText("Enter Street, Address, Concordia Classroom...")))
+
+        //Checks if Autocomplete options for text "h" are displayed
+        val searchAutoComplete = onView(
+            allOf(
+                withId(R.id.search_src_text), isDisplayed()
+            )
+        )
+        searchAutoComplete.perform(ViewActions.typeText("h"), ViewActions.closeSoftKeyboard())
+
+        Thread.sleep(2000)
+
+        //Checks if room suggestion H-400 is displayed and performs a click on the search card
+        val searchCardView = onView(
+            allOf(
+                withId(R.id.search_suggestions_card_view),
+                withChild(
+                    allOf(
+                        withId(R.id.relative_layout1),
+                        withChild(
+                            allOf(
+                                withId(R.id.search_suggestion),
+                                withText("H-400")
+                            )
+                        )
+                    )
+                ),
+                isDisplayed()
+            )
+        )
+
+        searchCardView.perform(click())
+
+        //Performs click on the Set Navigation Button
+        val setNavImageButton = onView(
+            allOf(
+                withId(R.id.setNavigationPoint),
+                withContentDescription("SetNavigation"),
+                hasSibling(
+                    allOf(
+                        withId(R.id.relative_layout1),
+                        withChild(
+                            allOf(
+                                withId(R.id.search_suggestion),
+                                withText("H-400")
+                            )
+                        )
+                    )
+                ),
+                isDisplayed()
+            )
+        )
+
+        setNavImageButton.perform(click())
+
+        //Checks if back button is displayed
+        val backImageButton = onView(
+            allOf(
+                withId(R.id.backFromNavigationButton), withContentDescription("Back"),
+                isDisplayed()
+            )
+        )
+        backImageButton.check(matches(isDisplayed()))
+
+        //Checks if The "From" field displays text "From"
+        val fromText = onView(
+            allOf(
+                withId(R.id.search_src_text), withHint("From"),
+                isDisplayed()
+            )
+        )
+        fromText.check(matches(withHint("From")))
+
+        //Checks that if a Clear button is displayed next to the search field
+        val clearButtonImageView = onView(
+            allOf(
+                withId(R.id.search_close_btn), withContentDescription("Clear query"),
+                isDisplayed()
+            )
+        )
+        clearButtonImageView.check(matches(isDisplayed()))
```

Comment:
As mentioned earlier, avoid declaring variable for onView method specially since it is not being reused.